R/html_dependency.R

```diff
@@ -505,6 +518,68 @@ renderDependencies <- function(dependencies,
   HTML(paste(html, collapse = "\n"))
 }
 
+
+
+renderScript <- function(script, srcpath, encodeFunc, hrefFilter) {
+  leaf_fields <- c("src", "integrity", "crossorigin")
+
+  check_names <- function(script) {
+    if (length(names(script)) != 0) {
+      if (!all(names(script) %in% leaf_fields)) {
+        invalid_names <- names(script)[!names(script) %in% leaf_fields]
+        stop(
+          "Providing ", paste(invalid_names, collapse = T), " not supported",
+          " for <script> tags")
+      }
+
+      if (!hasName(script, "src")) {
+        stop("Named script elements must provide src")
+      }
+    }
+  }
+
+  check_names(script)
+
+  if (length(names(script)) > 0) script <- list(script)
```

Comment:
I think the algorithm can go something like this:

* If the input is all unnamed, like `c("a.js", "b.js")`, convert to `list(list(src="a.js"), list(src="b.js"))` with something like:
    ```R
    lapply(script, function(x) list(src = x))
    ```
* If all named, skip that step.
* If it’s a mix of named and unnamed, throw an error. 


Keep in mind that vectors/lists can have names, but the names are empty. For example:

```R
# Named list with empty names:
x <- list(a=1, 2, 3)[2:3]
names(x)
#> [1] "" ""

# Compare to truly unnamed list
names(list(2, 3))
#> NULL
```

Note that in Shiny, we have functions that take this into account. It might make sense to bring those (or similar functions) into htmltools.
https://github.com/rstudio/shiny/blob/7aa3a243ba7ee33c89060a52cb7518e892421ecf/R/utils.R#L147-L174

Reply:
Would you like me to do so in this PR?

Reply:
I've added in the two mentioned utils to `utils.R`.

It accepts a named list of characters, or an unamed list of lists/characters. I forgot to throw if it's a mix (oops)